In [78]:
#Importar las librerias necesarias
import pandas as pd
import numpy as np

In [98]:
# Cargar el dataset
data=pd.read_csv("AccidentesAviones.csv")

In [99]:
# Explorar el dataset
# Visualizar las primeras filas del DataFrame
data.head(3)

ID               fecha HORA_declara                                Ruta   
0   0  September 17, 1908         1718                 Fort Myer, Virginia  \
1   1  September 07, 1909            ?             Juvisy-sur-Orge, France   
2   2       July 12, 1912         0630           Atlantic City, New Jersey   
3   3     August 06, 1913            ?  Victoria, British Columbia, Canada   
4   4  September 09, 1913         1830                  Over the North Sea   

                 OperadOR flight_no          route                 ac_type   
0    Military - U.S. Army         ?  Demonstration        Wright Flyer III  \
1                       ?         ?       Air show          Wright Byplane   
2    Military - U.S. Navy         ?    Test flight               Dirigible   
3                 Private         ?              ?        Curtiss seaplane   
4  Military - German Navy         ?              ?  Zeppelin L-1 (airship)   

  registration cn_ln all_aboard PASAJEROS_A_BORDO crew_aboard   
0            ?     1          2                 1           1  \
1          SC1     ?          1                 0           1   
2            ?     ?          5                 0           5   
3            ?     ?          1                 0           1   
4            ?     ?         20                 ?           ?   

  cantidad_de_fallecidos passenger_fatalities crew_fatalities ground   
0                      1                    1               0      0  \
1                      1                    0               0      0   
2                      5                    0               5      0   
3                      1                    0               1      0   
4                     14                    ?               ?      0   

                                             summary  
0  During a demonstration flight, a U.S. Army fly...  
1  Eugene Lefebvre was the first pilot to ever be...  
2  First U.S. dirigible Akron exploded just offsh...  
3  The first fatal airplane accident in Canada oc...  
4  The airship flew into a thunderstorm and encou...

In [100]:
# Obtener información del dataset
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5008 entries, 0 to 5007
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   ID                      5008 non-null   int64 
 1   fecha                   5008 non-null   object
 2   HORA_declara            5008 non-null   object
 3   Ruta                    5008 non-null   object
 4   OperadOR                5008 non-null   object
 5   flight_no               5008 non-null   object
 6   route                   5007 non-null   object
 7   ac_type                 5008 non-null   object
 8   registration            5008 non-null   object
 9   cn_ln                   5008 non-null   object
 10  all_aboard              5008 non-null   object
 11  PASAJEROS_A_BORDO       5008 non-null   object
 12  crew_aboard             5008 non-null   object
 13  cantidad_de_fallecidos  5008 non-null   object
 14  passenger_fatalities    5008 non-null   object
 15  crew

In [101]:
# Verificar duplicados
data.duplicated().value_counts()

False    5008
Name: count, dtype: int64

In [102]:
# Seleccionar las columnas para el análisis univariado
columnas = ["ID",'fecha', 'HORA_declara', 'Ruta', 'OperadOR', 'flight_no', 'route',
            'ac_type', 'registration', 'cn_ln', 'all_aboard', 'PASAJEROS_A_BORDO',
            'crew_aboard', 'cantidad_de_fallecidos', 'passenger_fatalities',
            'crew_fatalities', 'ground', 'summary']

# Obtener el resumen estadístico de las columnas numéricas
resumen_estadistico = data[columnas].describe()

# Calcular la moda de las columnas categóricas
moda = data[columnas].mode()

# Calcular la tabla de frecuencia para las columnas categóricas
tabla_frecuencia = pd.DataFrame()
for col in columnas:
    if data[col].dtype == 'object':  # Considerar solo las columnas categóricas
        tabla_frecuencia[col] = data[col].value_counts()

# Mostrar los resultados
print("Resumen estadístico:")
print(resumen_estadistico)
print("\nModa:")
print(moda)
print("\nTabla de frecuencia:")
print(tabla_frecuencia)


Resumen estadístico:
                ID
count  5008.000000
mean   2503.500000
std    1445.829404
min       0.000000
25%    1251.750000
50%    2503.500000
75%    3755.250000
max    5007.000000

Moda:
        ID               fecha HORA_declara            Ruta  OperadOR   
0        0     August 31, 1988            ?  Moscow, Russia  Aeroflot  \
1        1  September 11, 2001          NaN             NaN       NaN   
2        2                 NaN          NaN             NaN       NaN   
3        3                 NaN          NaN             NaN       NaN   
4        4                 NaN          NaN             NaN       NaN   
...    ...                 ...          ...             ...       ...   
5003  5003                 NaN          NaN             NaN       NaN   
5004  5004                 NaN          NaN             NaN       NaN   
5005  5005                 NaN          NaN             NaN       NaN   
5006  5006                 NaN          NaN             NaN       NaN  

In [109]:
#En el analisis exploratorio no se deberia modificar el dataset, solo observarlo, pero para fines de exploracion de los datos se reemplaza los "?" por valores nulos y asi poder aplicar diferentes herramientas de exploración.
columnas=['ID', 'fecha', 'HORA_declara', 'Ruta', 'OperadOR', 'flight_no', 'route',
       'ac_type', 'registration', 'cn_ln', 'all_aboard', 'PASAJEROS_A_BORDO',
       'crew_aboard', 'cantidad_de_fallecidos', 'passenger_fatalities',
       'crew_fatalities', 'ground', 'summary']
# Reemplazar los valores no numéricos por NaN
data[columnas] = data[columnas].replace('?', np.nan)

In [110]:
# Seleccionar las columnas numéricas de interés
columnas_numericas = ['all_aboard', 'PASAJEROS_A_BORDO', 'crew_aboard', 'cantidad_de_fallecidos', "passenger_fatalities","crew_fatalities",  "ground"]

# Crear un nuevo DataFrame con las columnas numéricas
data_numericas = data[columnas_numericas]

# Convertir las columnas a tipo numérico
data_numericas = data_numericas.astype(float)

# Calcular la matriz de correlación para las columnas numéricas
correlation_matrix = data_numericas.corr()

# Mostrar la matriz de correlación
print("Matriz de correlación:")
print(correlation_matrix)


Matriz de correlación:
                        all_aboard  PASAJEROS_A_BORDO  crew_aboard   
all_aboard                1.000000           0.997487     0.568893  \
PASAJEROS_A_BORDO         0.997487           1.000000     0.510741   
crew_aboard               0.568893           0.510741     1.000000   
cantidad_de_fallecidos    0.739646           0.738965     0.441561   
passenger_fatalities      0.746712           0.747417     0.395682   
crew_fatalities           0.320119           0.274408     0.720937   
ground                    0.020327           0.018833     0.032121   

                        cantidad_de_fallecidos  passenger_fatalities   
all_aboard                            0.739646              0.746712  \
PASAJEROS_A_BORDO                     0.738965              0.747417   
crew_aboard                           0.441561              0.395682   
cantidad_de_fallecidos                1.000000              0.992972   
passenger_fatalities                  0.992972          

## CONCLUSIONES

### Según la información proporcionada por data.info(), podemos sacar las siguientes conclusiones:

El conjunto de datos tiene 5008 filas y 18 columnas.
La mayoría de las columnas son de tipo object, lo que indica que contienen datos de tipo texto o mixto.
La columna "ID" es la única columna numérica en el conjunto de datos y se representa como tipo int64.
No hay valores nulos en ninguna de las columnas, ya que todas las columnas tienen 5008 valores no nulos.

El conjunto de datos parece estar relativamente limpio, sin valores nulos en ninguna de las columnas. Sin embargo, es importante tener en cuenta que algunos de los valores que se muestran como no nulos pueden contener datos faltantes representados por caracteres como "?".

Valores nulos (?): Se observa que hay varias columnas con valores nulos. La cantidad de valores nulos varía para cada columna. Para lidiar con estos valores nulos, se pueden considerar opciones como eliminar las filas o columnas con valores nulos, imputar los valores faltantes o trabajar con las filas que tienen datos completos.

No hay datos duplicados en el conjunto de datos. Todos los 5008 registros son únicos. Cada fila en el conjunto de datos representa una instancia distinta, sin duplicados. 

### Basado en el resumen estadístico y la tabla de frecuencia proporcionados, podemos extraer las siguientes conclusiones:

1. ID: La columna ID contiene 5008 valores únicos que van desde 0 hasta 5007. No hay datos duplicados.

2. Fecha: La columna Fecha tiene una amplia variedad de valores únicos, y la fecha "August 31, 1988" y "September 11, 2001" son las fechas más frecuentes, con 4 ocurrencias cada una.

3. HORA_declara: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

4. Ruta: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

5. OperadOR: La aerolínea "Aeroflot" es la más frecuente en la columna OperadOR.

6. flight_no: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

7. route: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

8. ac_type: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

9. registration: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

10. cn_ln: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

11. all_aboard: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

12. PASAJEROS_A_BORDO: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

13. crew_aboard: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

14. cantidad_de_fallecidos: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

15. passenger_fatalities: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

16. crew_fatalities: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

17. ground: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

18. summary: No se proporcionan estadísticas ni frecuencias para esta columna, lo que sugiere que podría haber valores faltantes o datos no numéricos.

### Basándonos en la matriz de correlación que has proporcionado, podemos obtener las siguientes conclusiones:

La cantidad de pasajeros a bordo (PASAJEROS_A_BORDO) y el número total de personas a bordo (all_aboard) tienen una correlación muy alta de 0.997487. Esto indica que estos dos atributos están fuertemente relacionados y tienden a aumentar o disminuir juntos.

La cantidad de tripulación a bordo (crew_aboard) también muestra una correlación positiva con el número total de personas a bordo (all_aboard) y la cantidad de pasajeros a bordo (PASAJEROS_A_BORDO), aunque en menor medida.

La cantidad de fallecidos (cantidad_de_fallecidos) muestra una correlación significativa con el número total de personas a bordo (all_aboard) y la cantidad de pasajeros a bordo (PASAJEROS_A_BORDO). Esto sugiere que un mayor número de personas a bordo puede estar relacionado con un mayor número de fallecidos en un accidente.

La cantidad de fallecidos (cantidad_de_fallecidos) y las fatalidades de pasajeros (passenger_fatalities) tienen una correlación muy alta de 0.992972. Esto indica que estos dos atributos están fuertemente relacionados y tienden a variar juntos.

Las fatalidades de tripulación (crew_fatalities) también muestran una correlación positiva con la cantidad de fallecidos (cantidad_de_fallecidos) y las fatalidades de pasajeros (passenger_fatalities). Sin embargo, esta correlación es menor en comparación con las correlaciones anteriores.

La correlación entre los atributos numéricos y los fallecidos en tierra (ground) es muy baja, lo que indica que los fallecidos en tierra no está fuertemente relacionado con los otros atributos numéricos en la matriz.